documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-request-customizers.adoc

```diff
@@ -0,0 +1,112 @@
+`HostHeaderCustomizer` synthesizes the `Host` HTTP header for such requests, so that WebSocket web applications that inspect the initial HTTP request before the WebSocket communication will work seamlessly in any HTTP protocol version.
+
+For more information about how to configure `HostHeaderCustomizer`, see also link:{javadoc-url}/org/eclipse/jetty/server/HostHeaderCustomizer.html[the javadocs].
+
+[[pg-server-http-request-customizer-proxy]]
+===== `ProxyCustomizer`
+
+`ProxyCustomizer` should be added when Jetty receives requests from a reverse proxy on behalf of a remote client, prefixed by the PROXY protocol (see also this section about the xref:pg-server-http-connector-protocol-proxy-http11[PROXY protocol]).
+
+`ProxyCustomizer` adds the reverse proxy IP address and port as `Request` attributes.
+Web applications may use these attributes in conjunction with the data exposed by `ForwardedRequestCustomizer` (see xref:pg-server-http-request-customizer-forwarded[this section]).
+
+For more information about how to configure `ProxyCustomizer`, see also link:{javadoc-url}/org/eclipse/jetty/server/ProxyCustomizer.html[the javadocs].
+
+[[pg-server-http-request-customizer-rewrite]]
+===== `RewriteCustomizer`
+
+`RewriteCustomizer` is similar to `RewriteHandler` (see xref:pg-server-http-handler-use-rewrite[this section]), but a `RewriteCustomizer` cannot send a response or otherwise complete the request/response processing.
+
+A `RewriteCustomizer` is mostly useful if you want to rewrite the request URI _before_ the `Handler` chain is invoked.
+However, a very similar effect can be achieved by having the `RewriteHandler` as the first `Handler` (the child `Handler` of the `Server` instance).
+
+Since `RewriteCustomizer` cannot send a response or complete the request/response processing, ``Rule``s that do so such as redirect rules have no effect and are ignored; only ``Rule``s that modify or wrap the `Request` will have effect and be applied.
+
+Due to this limitation, it is often a better choice to use `RewriteHandler` instead of `RewriteCustomizer`.
+
+For more information about how to configure `RewriteCustomizer`, see also link:{javadoc-url}/org/eclipse/jetty/rewrite/RewriteCustomizer.html[the javadocs].
+
+[[pg-server-http-request-customizer-secure]]
+===== `SecureRequestCustomizer`
+
+`SecureRequestCustomizer` should be added when Jetty receives requests over a secure `Connector`.
+
+`SecureRequestCustomizer` adds TLS information as request attributes, in particular an instance of `EndPoint.SslSessionData` that contains information about the negotiated TLS cipher suite and possibly client certificates, and an instance of `org.eclipse.jetty.util.ssl.X509` that contains information about the server certificate.
+
+`SecureRequestCustomizer` also adds, if configured so, the `Strict-Transport-Security` HTTP response header (for more information about this header, see link:https://datatracker.ietf.org/doc/html/rfc6797[its specification]).
```

Comment:
We should modify that class to use a PersistentHttpField for the STS response header, so it survives any app reset

Reply:
I'd leave it so for now. If you really want that, we need to make `PersistentHttpField` more public, as now cannot be used outside the `jetty-server` module.

Reply:
Perhaps we should keep PersistentHttpField internal, but after customization convert all added fields to Persistent ones.  That way the user never need know about it and no documentation required.   Different PR.